### PR TITLE
DTSERWFOUR-387-edit-venmo account partial edit is not successful

### PR DESCRIPTION
**Pull request by shanavas123**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -24,7 +24,7 @@ allprojects {
 
     }
 
-    project.version = "1.0.0-beta09"
+    project.version = "1.0.0-beta09-SNAPSHOT"
 
 }
```
Comment (malaw-hw) [inline build.gradle:29]: this should be 1.0.0-beta10-SNAPSHOT,
as we had mentioned, we have released "1.0.0-beta09" already
https://mvnrepository.com/artifact/com.hyperwallet.android.ui/userrepository.  please update the version for other component's grade file as well.

**Review by malaw-hw — CHANGES_REQUESTED** [review]

**Review by malaw-hw — DISMISSED** [review]

**Review by pdamodarannair — DISMISSED** [review]

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/ExpiryDateWidget.java [hunk 2]
```diff
@@ -186,6 +186,7 @@ public void afterTextChanged(Editable s) {
             editText.setText(mExpireDateUtils.convertDateFromServerFormat(mDefaultValue));
 
             editText.setOnKeyListener(new DefaultKeyListener(mDefaultFocusView, editText));
+            editText.addTextChangedListener(new TextChangeListener());
```
Comment (vwagh-hw) [inline transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/ExpiryDateWidget.java:189]: ExpiryDateWidget already has a `TextChangedListener` why assign a new one?

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/NumberWidget.java [hunk 3]
```diff
@@ -83,6 +83,7 @@ public void onFocusChange(View v, boolean hasFocus) {
             }
 
             editText.setOnKeyListener(new DefaultKeyListener(mDefaultFocusView, editText));
+            editText.addTextChangedListener(new TextChangeListener());
```
Comment (vwagh-hw) [inline transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/NumberWidget.java:86]: NumberWidget already has a `TextChangedListener` why assign a new one?

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/PhoneWidget.java [hunk 4]
```diff
@@ -76,6 +76,7 @@ public void onFocusChange(View v, boolean hasFocus) {
             editText.setText(mDefaultValue);
             editText.setInputType(InputType.TYPE_CLASS_PHONE);
             editText.setOnKeyListener(new DefaultKeyListener(mDefaultFocusView, editText));
+            editText.addTextChangedListener(new TextChangeListener());
```
Comment (vwagh-hw) [inline transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/PhoneWidget.java:79]: PhoneWidget already has a `TextChangedListener`

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/TextWidget.java [hunk 5]
```diff
@@ -75,6 +75,7 @@ public void onFocusChange(View v, boolean hasFocus) {
             editText.setText(mDefaultValue);
             editText.setInputType(InputType.TYPE_CLASS_TEXT);
             editText.setOnKeyListener(new DefaultKeyListener(mDefaultFocusView, editText));
+            editText.addTextChangedListener(new TextChangeListener());
```
Comment (vwagh-hw) [inline transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/TextWidget.java:78]: There is already a TextChangedListener for TextWidget. Why not use the existing one?

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/ExpiryDateWidget.java (hunk 2 above)
Comment (shanavas123) [reply transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/ExpiryDateWidget.java:189]: https://developer.android.com/reference/android/view/View.OnKeyListener
OnKeyListener not triggered in some keyboard, google recommended to use textchange listener for text changes.

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/NumberWidget.java (hunk 3 above)
Comment (shanavas123) [reply transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/NumberWidget.java:86]: https://developer.android.com/reference/android/view/View.OnKeyListener
OnKeyListener not triggered in some keyboard, google recommended to use textchange listener for text changes.

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/TextWidget.java (hunk 5 above)
Comment (shanavas123) [reply transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/TextWidget.java:78]: https://developer.android.com/reference/android/view/View.OnKeyListener
OnKeyListener not triggered in some keyboard, google recommended to use textchange listener for text changes.

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/PhoneWidget.java (hunk 4 above)
Comment (shanavas123) [reply transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/PhoneWidget.java:79]: https://developer.android.com/reference/android/view/View.OnKeyListener
OnKeyListener not triggered in some keyboard, google recommended to use textchange listener for text changes.

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/ExpiryDateWidget.java (hunk 2 above)
Comment (shanavas123) [reply transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/ExpiryDateWidget.java:189]: Screenshot 2021-02-26 at 2 23 10 PM

Comment (shanavas123) [reply transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/ExpiryDateWidget.java]: fixed

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/NumberWidget.java (hunk 3 above)
Comment (shanavas123) [reply transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/NumberWidget.java]: fixed

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/PhoneWidget.java (hunk 4 above)
Comment (shanavas123) [reply transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/PhoneWidget.java]: fixed

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/TextWidget.java (hunk 5 above)
Comment (shanavas123) [reply transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/TextWidget.java]: fixed

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/AbstractMaskedInputWidget.java [hunk 6]
```diff
@@ -188,6 +188,10 @@ public void beforeTextChanged(CharSequence s, int start, int count, int after) {
 
         @Override
         public void afterTextChanged(Editable s) {
+            if(mEditText.hasFocus())
+            {
+                isEdited = true;
+            }
```
Comment (vwagh-hw) [inline transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/AbstractMaskedInputWidget.java:194]: fix the formatting

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/AbstractMaskedInputWidget.java [hunk 7]
```diff
@@ -233,6 +237,10 @@ public void afterTextChanged(Editable s) {
                     mEditText.addTextChangedListener(this);
                 }
             }
+            if(mEditText.hasFocus())
+            {
+                isEdited = true;
+            }
```
Comment (vwagh-hw) [inline transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/AbstractMaskedInputWidget.java:243]: fix formatting

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/ExpiryDateWidget.java [hunk 8]
```diff
@@ -178,6 +178,10 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
 
                 @Override
                 public void afterTextChanged(Editable s) {
+                    if(editText.hasFocus())
+                    {
+                        isEdited = true;
+                    }
```
Comment (vwagh-hw) [inline transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/ExpiryDateWidget.java:184]: fix formatting

**Review by vwagh-hw — COMMENTED** [review]
Code has changed a lot, this should be tested again @SaiNarendran

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/AbstractMaskedInputWidget.java (hunk 6 above)
Comment (shanavas123) [reply transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/AbstractMaskedInputWidget.java]: fixed

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/AbstractMaskedInputWidget.java (hunk 7 above)
Comment (shanavas123) [reply transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/AbstractMaskedInputWidget.java]: fixed

File: transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/ExpiryDateWidget.java (hunk 8 above)
Comment (shanavas123) [reply transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/ExpiryDateWidget.java]: fixed

**Comment by SaiNarendran** [comment]
> Code has changed a lot, this should be tested again @SaiNarendran

@vwagh-hw - QA has completed another round of validation and it looks fine.

**Review by vwagh-hw — APPROVED** [review]

**Review by pdamodarannair — APPROVED** [review]

**Review by SaiNarendran — APPROVED** [review]

**Pull request merged**